File: frontend/src/lib/fastapi/quote-extraction.ts
import { apiClient } from "./client";

export interface QuoteItem {
  text: string;
  reason?: string;
}

export interface QuoteExtractionResponse {
  success: boolean;
  quotes?: QuoteItem[];
  error?: string;
  detail?: string;
}

export interface QuoteExtractionRequest {
  transcript: string;
  style: "punchy" | "insightful" | "contrarian" | "emotional";
  count: number;
}

export async function extractQuotes(
  request: QuoteExtractionRequest,
): Promise<QuoteExtractionResponse> {
  return apiClient.request<QuoteExtractionResponse>("/v1/quote-extraction", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(request),
  });
}

File: backend/app/api/v1/quote_extraction.py
from fastapi import APIRouter
from pydantic import BaseModel, Field
from typing import List, Optional, Literal

from ...agents.quote_extraction.extractor import extract_quotes

router = APIRouter(prefix="/quote-extraction")


class QuoteExtractionRequest(BaseModel):
    transcript: str
    style: Literal["general", "punchy", "insightful", "contrarian", "emotional"] = "general"
    count: int = Field(10, ge=1, le=30)


class QuoteItem(BaseModel):
    text: str
    reason: Optional[str] = None


class QuoteExtractionResponse(BaseModel):
    success: bool
    quotes: Optional[List[QuoteItem]] = None
    error: Optional[str] = None


@router.post("", response_model=QuoteExtractionResponse)
async def extract_quotes_endpoint(request: QuoteExtractionRequest):
    try:
        quotes = await extract_quotes(
            transcript=request.transcript,
            style=request.style,
            count=request.count,
        )
        return QuoteExtractionResponse(success=True, quotes=quotes)
    except Exception as exc:
        return QuoteExtractionResponse(success=False, error=str(exc))
